runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9639,6 +9639,8 @@ TR::CompilationInfoPerThreadBase::compile(
 
          // FAR: should we do postpone this copying until after CHTable commit?
          metaData->runtimeAssumptionList = *(compiler->getMetadataAssumptionList());
+         if (metaData->runtimeAssumptionList)
+            reinterpret_cast<TR::SentinelRuntimeAssumption *>(metaData->runtimeAssumptionList)->setOwningMetadata(metaData);
```

Comment:
Is the owning metadata of a runtime assumption list mutable? If not, then to me it makes most sense for this to be mandatory to be initialized in the constructor of the object at the point of creation. This assuming we have the owning metadata pointer available at that point.

Reply:
The owning metadata is not mutable, however we don't have metadata available at the moment of sentinel creation.
The sentinel is normally created in the compilation object constructor, and the assumption list is populated during compilation and CH table commit.
While metadata only becomes available once the compilation is over and the relocation phase begins, and it will not be created if a compilation fails before that stage.

Reply:
Ok, so then this is not possible. I think @dsouzai's comment of setting it at the point of metadata creation is the next best thing.